test/methods.jl

```diff
@@ -126,6 +126,26 @@ end
     @test_throws ArgumentError combine()
 end
 
+@testset "dark subtraction" begin
+    # testing non-mutating version
+    @test @inferred(subtract_dark(ones(5, 5), ones(5, 5))) == zeros(5, 5)
+    @test @inferred(subtract_dark(ones(5, 5), ones(5, 5), data_exposure = 1, dark_exposure = 4, scale = true)) == fill(0.75, 5, 5)
+
+    # testing mutating version
+    frame = ones(5, 5)
+    dark_frame = ones(5, 5)
+    subtract_dark!(frame, dark_frame)
+    @test frame ≈ zeros(5, 5)
```

Comment:
@giordano @miles if the above line is changed to `@test frame == zeros(5, 5)`, it fails. Any idea why is it so?

Reply:
It might be from floating point round off

Reply:
Why does it fail for you?
```julia
julia> frame = ones(5, 5)
5×5 Array{Float64,2}:
 1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0

julia> dark_frame = ones(5, 5)
5×5 Array{Float64,2}:
 1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0
 1.0  1.0  1.0  1.0  1.0

julia> subtract_dark!(frame, dark_frame)
5×5 Array{Float64,2}:
 0.0  0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0  0.0

julia> frame == zeros(5, 5)
true
```

Reply:
I don't know what went wrong at that time, it is working fine right now!

Reply:
Yeah, I'd be surprised if you don't get zero when simply subtracting two identical matrices, without further operations